### DISCUSSION TITLE
What does inhibiting leading does in (sub)section?

### DISCUSSION THREAD

**Discussion by Omikhleia** [sile-typesetter/sile#1288] [Q&A] [answered]
Greetings,

The `\section` and `\subsection` in the (default) book class both do a `SILE.typesetter:inhibitLeading()` at their end. Chapters and parts do not.

I am trying to understand what it does, but couldn't make it out clearly. In my own class(es), I added it too to all my sectioning commands, but I am having weird behavior with it. It seems better looking _without_ it, but I have no idea of the actual impact generally. What does it do exactly and when is it expected? Unless mistaken, it is only used there (i.e. section and subsection), so I assume there is a reason for it, but why only there then?

**Comment by alerque** [comment]
I believe this was an early way of setting `\noindent`, but these should be updated to actually use `SILE.call("noindent")` so we don't fragment the implementations. It should just avoid the next paragraph after section headings being indented.

**Reply by jslabovitz** [reply]
I happened to run into this bug just now, while using the `book` class along with grid typesetting. The paragraphs after the section nor subsection heading are not aligned with the grid. 

I've been working on my own `book` subclass, so pasted the definitions for the headings in, and replaced `SILE.typesetter:inhibitLeading()` with `SILE.call("noindent")`, and got the expected grid alignment. I'm not even sure *that's* necessary, as both those commands call `noindent` earlier. I tried removing the `inhibitLeading` line entirely, and that seemed to work just as well.

Test code, using SILE v0.14.8 (Lua 5.4):

```
\begin[class=book,papersize="a5"]{document}
% \begin[class=johnl-book,papersize="a5"]{document}

\use[module="packages.grid"]
\grid[spacing="18pt"]
\grid:debug

\font[size="14pt"]

\chapter{A chapter}

foo bar zot

foo bar zot

\section{A section}

foo bar zot

foo bar zot

\section{A subsection}

foo bar zot

foo bar zot

\end{document}
```

**Reply by Omikhleia** [reply]
`SILE.typesetter:inhibitLeading()` bears no relation with `SILE.call("noindent")`.
The former cancels vertical leading, the latter cancels paragraph indentation. Two different things for two different purposes.
_This being said_, the mystery why the book class cancels vertical leading after sections and subsections (only) remains entire ;)

**Reply by alerque** [reply]
Didier is right of course, I must have been sleep deprived back in Dec '21. Also @Omikhleia do we have a cross reference bug number for that?

**Reply by Omikhleia** [reply]
@alerque The closest related bug reference we have on that topic is #1300 (with Simon intervening on the difference between the vertical leading inhibition vs. the paragraph indent - but he didn't provide insights why the former was used in section and subsection)

**Reply by Omikhleia** [reply, marked as answer]
For those interested, https://github.com/Omikhleia/resilient.sile/issues/27 - It's on a slightly different topic, but I also took that opportunity to show what inhibiting leading does (see the picture).

**Reply by Omikhleia** [reply]
Asking ChatGPT, "leading suppression" after header section is a matter of taste and typographic choice, without any strong rule or convention. I'll consider this is the answer.

As noted, in my own resilient.book class, I decided not to have this behavior.
As for SILE's default book class, I no longer use it, so let's say it may do it if it so wishes - whatever occurs then with other layout strategies (grid, etc.) is beyond the initial question I had in 2021.
